classes/class-gf-securesubmit.php

```diff
@@ -1,950 +1,1597 @@
-            }
-
-            $text = str_replace($mergeTag, $value, $text);
-        }
-        return $text;
-    }
-
-    public function addClientSideMergeTags($form)
-    {
-        include plugin_dir_path(__FILE__) . '../templates/client-side-merge-tags.php';
-        return $form;
-    } 
-    /**Attempts to get real ip even if there is a proxy chain
-     * @return string
-     */
-    private function getRemoteIP() {
-        $remoteIP = $_SERVER['REMOTE_ADDR'];
-        if (array_key_exists('HTTP_X_FORWARDED_FOR', $_SERVER) && $_SERVER['HTTP_X_FORWARDED_FOR'] != '') {
-            $remoteIPArray = array_values(
-                array_filter(
-                    explode(
-                        ',',
-                        $_SERVER['HTTP_X_FORWARDED_FOR']
-                    )
-                )
-            );
-            $remoteIP = end($remoteIPArray);
-        }
-        return $remoteIP;
-    }
-}
+<?php
+GFForms::include_payment_addon_framework();
+include_once 'class-gf-field-hpsach.php';
+
+/**
+ * Handles Heartlands Payments with Gravity Forms
+ * Class GFSecureSubmit
+ */
+class GFSecureSubmit
+    extends GFPaymentAddOn {
+    /**
+     * @var bool
+     */
+    private $isCC = null;
+    /**
+     * @var string
+     */
+    protected $_version = GF_SECURESUBMIT_VERSION;
+
+    /**
+     * @var string
+     */
+    protected $_min_gravityforms_version = '1.9.1.1';
+    /**
+     * @var string
+     */
+    protected $_slug = 'gravityforms-securesubmit';
+    /**
+     * @var string
+     */
+    protected $_path = 'gravityforms-securesubmit/gravityforms-securesubmit.php';
+    /**
+     * @var string
+     */
+    protected $_full_path = __FILE__;
+    /**
+     * @var string
+     */
+    protected $_title = 'Gravity Forms SecureSubmit Add-On';
+    /**
+     * @var string
+     */
+    protected $_short_title = 'SecureSubmit';
+    /**
+     * @var bool
+     */
+    protected $_requires_credit_card = true;
+    /**
+     * @var bool
+     */
+    protected $_supports_callbacks = true;
+    /**
+     * @var bool
+     */
+    protected $_enable_rg_autoupgrade = true;
+
+    // Permissions
+    /**
+     * @var string
+     */
+    protected $_capabilities_settings_page = 'gravityforms_securesubmit';
+    /**
+     * @var string
+     */
+    protected $_capabilities_form_settings = 'gravityforms_securesubmit';
+    /**
+     * @var string
+     */
+    protected $_capabilities_uninstall = 'gravityforms_securesubmit_uninstall';
+
+    //Members plugin integration
+    /**
+     * @var array
+     */
+    protected $_capabilities
+        = ['gravityforms_securesubmit',
+           'gravityforms_securesubmit_uninstall'];
+
+    /**
+     * @var null
+     */
+    private static $_instance = null;
+    /**
+     * @var null
+     */
+    public $transaction_response = null;
+
+    /**
+     * @return GFSecureSubmit|null
+     */
+    public static
+    function get_instance()
```

Comment:
@slogsdon I think we differ in preference here but i will update it across the code to comply